analytics-integrations/kahuna/src/main/java/com/segment/analytics/internal/integrations/KahunaIntegration.java

```diff
@@ -57,25 +84,64 @@
   @Override public void identify(IdentifyPayload identify) {
     super.identify(identify);
 
-    String username = identify.traits().username();
-    KahunaAnalytics.setUsernameAndEmail(isNullOrEmpty(username) ? identify.userId() : username,
-        identify.traits().email());
-
-    KahunaAnalytics.setUserCredential(USERNAME_KEY, identify.traits().username());
-    KahunaAnalytics.setUserCredential(EMAIL_KEY, identify.traits().email());
-    KahunaAnalytics.setUserCredential(FACEBOOK_KEY, identify.traits().getString("facebook"));
-    KahunaAnalytics.setUserCredential(TWITTER_KEY, identify.traits().getString("twitter"));
-    KahunaAnalytics.setUserCredential(LINKEDIN_KEY, identify.traits().getString("linkedin"));
-
-    KahunaAnalytics.setUserAttributes(identify.traits().toStringMap());
+    KahunaAnalytics.setUserCredential(USER_ID_KEY, identify.userId());
+
+    Traits identityTraits = identify.traits();
+    Map<String, String> userAttributes = new HashMap<>();
+    for (String key : identityTraits.keySet()) {
+      if (identityTraits.get(key) == null) {
+        continue; // Skip null value objects.
+      } else if (SUPPORTED_KAHUNA_CREDENTIALS.contains(key)) {
+        KahunaAnalytics.setUserCredential(key, identityTraits.getString(key));
+      } else {
+        // We'll track unsupported Kahuna User Credentials as Kahuna User Attributes instead
+        // and make sure to format any Date objects appropriately.
+        Object value = identityTraits.get(key);
+        if (value instanceof Date) {
+          userAttributes.put(key, getISO8601StringForDate((Date) value));
+        } else {
+          userAttributes.put(key, String.valueOf(value));
+        }
+      }
+    }
+    KahunaAnalytics.setUserAttributes(userAttributes);
   }
 
   @Override public void track(TrackPayload track) {
     super.track(track);
 
-    // Although not documented, Kahuna wants revenue in cents
-    KahunaAnalytics.trackEvent(track.event(), track.properties().getInt("quantity", 0),
-        (int) (track.properties().revenue() * 100));
+    if (track == null || TextUtils.isEmpty(track.event()))
+      return;
+
+    String kahunaEventName = track.event();
+
+    kahunaEcommerceHelper.process(track);
+
+    if (track.properties() == null) {
+      KahunaAnalytics.trackEvent(kahunaEventName);
+    } else {
+      Object revenueObject = track.properties().get(SEGMENT_REVENUE_KEY);
```

Comment:
Delete SEGMENT_REVENUE_KEY and replace this line with `int revenue = (int) (track.properties().revenue() * 100)`